AdSecCoreTests/FindCrackLoadFunctionTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿using AdSecCore.Functions;
+
+namespace AdSecCoreTests {
+  public class FindCrackLoadFunctionTests {
+    private readonly FindCrackLoadFunction function;
+    public FindCrackLoadFunctionTests() {
+      function = new FindCrackLoadFunction();
+    }
+
+    [Fact]
+    public void IsFxWhenX() {
+      Assert.True(FindCrackLoadFunction.IsFx("X"));
+    }
+
+    [Fact]
+    public void IsFxWhenXX() {
+      Assert.True(FindCrackLoadFunction.IsFx("XX"));
+    }
+
+    [Fact]
+    public void IsFxWhenFx() {
+      Assert.True(FindCrackLoadFunction.IsFx("Fx"));
+    }
+
+    [Fact]
+    public void IsFxWhenFXX() {
+      Assert.True(FindCrackLoadFunction.IsFx("FXX"));
+    }
+
+    [Fact]
+    public void IsNotFxWhenZZ() {
+      Assert.False(FindCrackLoadFunction.IsFx("ZZ"));
+    }
+
+    [Fact]
+    public void IsMyyWhenY() {
+      Assert.True(FindCrackLoadFunction.IsMyy("Y"));
+    }
+
+    [Fact]
+    public void IsMyyWhenYY() {
+      Assert.True(FindCrackLoadFunction.IsMyy("YY"));
+    }
+
+    [Fact]
+    public void IsMyyWhenMy() {
+      Assert.True(FindCrackLoadFunction.IsMyy("MY"));
+    }
+
+    [Fact]
+    public void IsMyyWhenMyy() {
+      Assert.True(FindCrackLoadFunction.IsMyy("MYY"));
+    }
+
+    [Fact]
+    public void IsNotMyyWhenZZ() {
+      Assert.False(FindCrackLoadFunction.IsMyy("ZZ"));
+    }
+
+    [Fact]
+    public void IsMzzWhenZ() {
+      Assert.True(FindCrackLoadFunction.IsMzz("Z"));
+    }
+
+    [Fact]
+    public void IsMzzWhenZZ() {
+      Assert.True(FindCrackLoadFunction.IsMzz("ZZ"));
+    }
+
+    [Fact]
+    public void IsMzzWhenMz() {
+      Assert.True(FindCrackLoadFunction.IsMzz("MZ"));
+    }
+
+    [Fact]
+    public void IsMzzWhenMzz() {
+      Assert.True(FindCrackLoadFunction.IsMzz("MZZ"));
+    }
+
+    [Fact]
+    public void IsNotMzzWhenYY() {
+      Assert.False(FindCrackLoadFunction.IsMzz("YY"));
+    }
```

Comment:
those inputs should be done to the component itself.

```cs
function.OptimisedLoad.Value = "X";
function.Compute();
// Is there a check we can add?
```

if this is not possible...make all the small functions internal and then test it like so

```cs
    [Theory]
    [InlineData("X")]
    [InlineData("XX")]
    // ....
    public void FxShouldReturnTrueFor(string x) {
      Assert.True(FindCrackLoadFunction.IsFx(x));
    }

```